prov/psm3/src/psmx3_cq.c

```diff
@@ -765,9 +765,12 @@ psmx3_mq_status_copy(struct psm2_mq_req_user *req, void *status_array, int entry
 			op_context = sendv_rep->user_context;
 			buf = sendv_rep->buf;
 			flags |= sendv_rep->comp_flag;
+			data = PSMX3_GET_CQDATA(sendv_rep->tag);
```

Comment:
Oh, you mean the `if-`condition below. Yeah, that makes sense. I just imitated the previous code patterns. I'll let @acgoldma decide.

Reply:
@acgoldma asked me to take a look at this; I think @shefty is correct - move line 768 and put it between 769 and 770. (Adding the appropriate braces on line 769 and 770... ;-) )

Reply:
That is fine. My question is, for consistency reasons, should I do that for all instances? --
https://github.com/ofiwg/libfabric/blob/9cc1cf74e5a886647e1846d58744c14fa63c4979/prov/psm3/src/psmx3_cq.c#L439-L441
https://github.com/ofiwg/libfabric/blob/9cc1cf74e5a886647e1846d58744c14fa63c4979/prov/psm3/src/psmx3_cq.c#L461-L463
https://github.com/ofiwg/libfabric/blob/9cc1cf74e5a886647e1846d58744c14fa63c4979/prov/psm3/src/psmx3_cq.c#L483-L485
https://github.com/ofiwg/libfabric/blob/9cc1cf74e5a886647e1846d58744c14fa63c4979/prov/psm3/src/psmx3_cq.c#L505-L507
https://github.com/ofiwg/libfabric/blob/9cc1cf74e5a886647e1846d58744c14fa63c4979/prov/psm3/src/psmx3_cq.c#L622-L624

Reply:
Ah. Sorry, I see your issue now. So, you're actually changing code that has persisted since at least when psm >2< was first developed, and I'm concerned that changing it in all such places might have more side-effects than either of us realize, but I'm also concerned that it could end up getting set to a non-zero value (in the change you want to make) which might also have side-effects.

Which MPI are you running, or are you using libfabric directly?

Reply:
We are MPICH developers. MPICH is using libfabric directly.

Reading the man page on `fi_cq`, it does say that we are supposed ignore `data` if `FI_REMOTE_CQ_DATA` is not set. However, in our current code we don't actually check the completion flag, but always assumed the data is there. Part of the reason is for the path we use CQ data, we kind of depend on CQ data being sent. It'll be an error anyway if we send CQ data but couldn't receive it. Therefore, I believe move the data assignment into the `if-`branches to ensure consistency should be generally preferred. I'll add to the commit if you give me the go-ahead.

> I'm concerned that changing it in all such places might have more side-effects than either of us realize ...

Can we trust the tests? :)

Reply:
We just switched to `psm3` in our testing a couple of weeks ago. Before that, we are using `sockets`. I'll check with Intel (Texas) team. They routinely test on `psm2` and I haven't heard any issue from them.

Reply:
[@hzhou](https://github.com/hzhou) - you may also want to look at OpenMPI and check with some of the Intel MPICH experts on how this is handled by those MPIs.  PSM3 support for OFI is very similar to PSM2 and both OpenMPI and Intel MPI have successfully worked over PSM2 for over 10 years, and more recently have been successfully running over PSM3 for the last few years.

Reply:
Other MPIs may not be using CQ data.

Reply:
I'll add a patch to `psm2` later

Reply:
The question of whether to set `data` inside the branch was introduced here: https://github.com/ofiwg/libfabric/pull/5863. From the comment, it overloads CQ data for the purpose of `sep_id` for SEP address lookup optimization. Thus, the pattern was on purpose.

Reply:
@j-xiong Please review the `psm2` fix.

Reply:
@mwheinz - Are the changes here okay?